### PR TITLE
Task #165254 chore: Native Add to linkedin Button - Linkdin Education Profile certificate upload/share

### PR DESCRIPTION
**Pull request by tushar33**

### CODE REVIEW

File: src/components/com_tjcertificate/administrator/controllers/certificate.php [hunk 1]
```diff
@@ -48,15 +48,22 @@ public function download()
 
 		$certificate    = TJCERT::Certificate();
 
+		$certificateObj = $certificate::validateCertificate($uniqueCertificateId);
+
+		// If $uniqueCertificateId is not valid then object is empty so need to handle error (CALL TO A MEMBER FUNCTION CANDOWNLOAD() ON BOOLEAN)
+		if (empty($certificateObj))
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/controllers/certificate.php:54]: `if (!$certificateObj)`

File: src/components/com_tjcertificate/administrator/libraries/certificate.php [hunk 2]
```diff
@@ -962,15 +963,54 @@ public static function canDownload($uniqueCertificateId)
...
+		$orgParam = '&' . $params->get('organization_info') . '=' . $params->get('organization_id_name');
+
+		// Get client data
+		$dispatcher = JDispatcher::getInstance();
+		PluginHelper::importPlugin('content');
+		$result = $dispatcher->trigger('getCertificateClientData', array($this->client_id, $this->client));
+		$clientData = $result[0];
+
+		$certificateTitle = $clientData->title ? $clientData->title : $siteName . ' ' . Text::_('COM_TJCERTIFICATE_CERTIFICATE_DETAIL_VIEW_HEAD');
+		$certificateUrl = JUri::root() . substr(Route::_($this->getURL('', false)), strlen(Uri::base(true)) + 1);
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/libraries/certificate.php:1009]: $urlOptions = array();
$urlOptions['absolute'] = true;
$certificateUrl = $this->getURL($urlOptions, false)

File: src/components/com_tjcertificate/site/views/certificate/tmpl/default.php [hunk 3]
```diff
@@ -154,12 +154,13 @@
...
-						<a id="copyurl" data-toggle="popover" data-placement="bottom"
-						data-alt-url="<?php echo Uri::getInstance()->toString();?>"
-						data-content="Copied!" class="tj-certificate-btn" type="button"
-						onclick="certificateImage.copyUrl('copyurl');">
-						<?php echo Text::_('COM_TJCERTIFICATE_CERTIFICATE_URL_COPY');?>
-						</a>
+						<?php if ($this->params->get('linkedin_profile_btn')) {
+						?>
+							<a href="<?php echo $this->linkedInProfileUrl;?>" target="_blank">
+							<img src="media/com_tjcertificate/images/buttons/en_US.png" alt="LinkedIn Add to Profile button">
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/views/certificate/tmpl/default.php:160]: have you included this image in the MR?

File: src/components/com_tjcertificate/site/views/certificate/view.html.php [hunk 4]
```diff
@@ -93,7 +95,8 @@ public function display($tpl = null)
 
 		$certificateUrl = 'index.php?option=com_tjcertificate&view=certificate&certificate=' . $this->certificate->unique_certificate_id;
 		$this->certificateUrl = Uri::root() . substr(Route::_($certificateUrl), strlen(Uri::base(true)) + 1);
-		$this->downloadPermission = $certificate::canDownload($this->certificate->unique_certificate_id);
+		$this->downloadPermission = $this->certificate->canDownload();
+		$this->linkedInProfileUrl = $this->certificate->getLinkedInProfileUrl();
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/views/certificate/view.html.php:99]: I think this should be conditional based on the global config

File: src/components/com_tjcertificate/administrator/libraries/certificate.php [hunk 5]
```diff
@@ -962,15 +963,54 @@ public static function canDownload($uniqueCertificateId)
...
 	}
+
+	/**
+	 * Method to get linkedIn add to profile url.
+	 *
+	 * @return  STRING
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function getLinkedInProfileUrl()
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/libraries/certificate.php:982]: getAddToLinkedInProfileUrl

File: src/components/com_tjcertificate/site/views/certificate/view.html.php (hunk 4 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/views/certificate/view.html.php:99]: done

File: src/components/com_tjcertificate/site/views/certificate/tmpl/default.php (hunk 3 above)
Comment (tushar33) [reply src/components/com_tjcertificate/site/views/certificate/tmpl/default.php:160]: now added

File: src/components/com_tjcertificate/administrator/libraries/certificate.php (hunk 2 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/libraries/certificate.php:1009]: done

File: src/components/com_tjcertificate/administrator/libraries/certificate.php (hunk 5 above)
Comment (tushar33) [reply src/components/com_tjcertificate/administrator/libraries/certificate.php:982]: done

**Pull request merged**